### ISSUE TITLE
Removing controls from parents

### ISSUE THREAD

**Issue by zevv** [andlabs/libui#41]
I'm trying to tear down an hierarchy of uiControls by recursively unparenting and destroying the child controls, all the way up to the window. 

```

        uiLabel *l = uiNewLabel("foo");
        uiWindow *w = uiNewWindow("win", 320, 200, 0);
        uiWindowSetChild(w, uiControl(l));

        uiControlSetParent(uiControl(l), NULL);
        uiControlDestroy(uiControl(l));
        uiControlDestroy(uiControl(w));
```

This results in the following abort:

```
** (lua:20378): CRITICAL **: [libui] common/control.c:94:uiControlVerifySetParent() POSSIBLE IMPLEMENTATION BUG; CONTACT ANDLABS:
attempt to double unparent uiControl 0x1309c60
Trace/breakpoint trap

```

What is the proper way to handle this?

Thank you,

**Comment by andlabs** [comment]
For now, call `uiWindowSetChild(w, NULL);` instead of `uiControlSetParent()` (that function is only for control implementations to use). I'm not sure if I want to keep allowing `NULL` as a parameter, but this is a good reason to allow it...

**Comment by zevv** [comment]
Ok, that makes sense. It is a bit of a hassle though, because to remove a control from the tree, you would need to know the type of the parent to be able to call the proper function (uiWindowSetChild, uiGroupSetChild). With boxes and tabs I'm not sure how to do this because it is only possible to remove a child by index, is that right?

What is the rationale for not allowing uiControlDestroy when the control still has a parent? Could this call simply be allowed and make sure the parent/child relationship is properly broken when a control is destroyed?

The reason I need this is to make libui work properly with garbage collection: when adding a child to a control I keep a separate administration with object references, so when the GC kicks in it will first start collection nodes which do not have any children, and then work it's way up to the root of the tree. If I would simply call Destroy on the top level window, libui would then destroy all the child objects, but the references to the GC will still exist. So my goal is to make sure each and every destroy is initiated by the GC only, and never done under the hood from libui directly.

**Comment by andlabs** [comment]
Merged with #328.
